packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js

```diff
@@ -15,25 +27,205 @@ const i18n = {
     }
 };
 
-const baseProps = {
-    smartGatewayBaseUrl: '/'
+const storeActions = {
+    loadPreferences: jest.fn(),
+    savePreferences: jest.fn(),
+    editPreference: jest.fn()
 };
 
-describe('ContactPreferences', () => {
-    it('should be defined', () => {
-        // Arrange
-        const propsData = {
-            ...baseProps
+const storeState = contactPreferencesViewModel;
+
+const mountContactPreferences = ({
+    actions = storeActions,
+    state = storeState,
+    mocks = sutMocks,
+    propsData = sutProps,
+    data = dataDefaults
+} = {}) => shallowMount(ContactPreferences, {
+    i18n,
+    localVue,
+    propsData,
+    data,
+    store: new Vuex.Store({
+        modules: {
+            fContactPreferencesModule: {
+                state,
+                actions,
+                namespaced: true
+            }
+        }
+    }),
+    mocks
+});
+
+describe('ContactPreferences Component', () => {
+    beforeEach(() => {
+        // Arrange & Act
+        dataDefaults = () => ({
+            isFormDirty: false,
+            showErrorPage: false
+        });
+        cookiesSpy = jest.fn();
+        httpSpy = jest.fn();
+        sutMocks = {
+            $parent: {
+                $emit: jest.fn()
+            },
+            $http: httpSpy,
+            $cookies: cookiesSpy
+        };
+        sutProps = {
+            authToken: token,
+            smartGatewayBaseUrl: baseUrl
         };
+        wrapper = mountContactPreferences();
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    describe('when mounting the component and calling `initialise`', () => {
+        it('should call the load action with the correct parameters', () => {
+            // Assert
+            expect(storeActions.loadPreferences).toHaveBeenCalledWith(expect.any(Object), {
+                api: {
+                    baseUrl,
+                    cookies: cookiesSpy,
+                    httpClient: httpSpy
+                },
+                authToken: token
+            });
+        });
+
+        it('should set showErrorPage flag to true if an error occurs', () => {
+            // Arrange
+            const errorActions = {
+                loadPreferences: jest.fn().mockImplementationOnce(() => {
+                    throw new Error('some-error');
+                })
+            };
+            wrapper = mountContactPreferences({ actions: errorActions });
```

Comment:
Nit: Does this meant the component is mounted twice for tests like this?

Reply:
It does but can remove the '_create wrapper_' from the `beforeEach()` and move into each test 👍